File: mycollab-web/src/main/java/com/esofthead/mycollab/module/crm/view/parameters/CallScreenData.java
/**
 * This file is part of mycollab-web.
 *
 * mycollab-web is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-web is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-web.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.crm.view.parameters;

import com.esofthead.mycollab.module.crm.domain.CallWithBLOBs;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class CallScreenData {
	public static class Add extends ScreenData<CallWithBLOBs> {

		public Add(CallWithBLOBs call) {
			super(call);
		}
	}

	public static class Edit extends ScreenData<CallWithBLOBs> {

		public Edit(CallWithBLOBs call) {
			super(call);
		}
	}
	
	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/java/com/mycollab/module/crm/view/parameters/CallScreenData.kt
package com.mycollab.module.crm.view.parameters

import com.mycollab.module.crm.domain.CallWithBLOBs
import com.mycollab.vaadin.mvp.ScreenData

object CallScreenData {
    class Add(call: CallWithBLOBs) : ScreenData<CallWithBLOBs>(call)

    class Edit(call: CallWithBLOBs) : ScreenData<CallWithBLOBs>(call)

    class Read(params: Int?) : ScreenData<Int>(params)
}
